### PR TITLE
Update python3Packages.{markupsafe,jinja2,itsdangerous,werkzeug,flask,click}

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Palletsproject's recent releases dropped Python 2 support. When do we?

I need these updates for home-assistant.

Flask:
- https://github.com/pallets/flask/releases/tag/2.0.1
- https://github.com/pallets/flask/releases/tag/2.0.0

Jinja2
- https://github.com/pallets/jinja/releases/tag/3.0.1
- https://github.com/pallets/jinja/releases/tag/3.0.0

Werkzeug
- https://github.com/pallets/werkzeug/releases/tag/2.0.1
- https://github.com/pallets/werkzeug/releases/tag/2.0.0

Click
- https://github.com/pallets/click/releases/tag/8.0.1
- https://github.com/pallets/click/releases/tag/8.0.0

itsdangerous
- https://github.com/pallets/itsdangerous/releases/tag/2.0.1
- https://github.com/pallets/itsdangerous/releases/tag/2.0.0

markupsafe
- https://github.com/pallets/markupsafe/releases/tag/2.0.1
- https://github.com/pallets/markupsafe/releases/tag/2.0.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/click/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/click/default.nix:16]: It doesn't look like you're using the scraped/published SHA256's from pypi.org: e.g. https://pypi.org/project/click/8.0.1/#copy-hash-modal-d0461d48-855f-4668-bdc8-8e742a09f958. Would you be able to use those instead? They should evaluate to the same, I guess Nix's standard representation is a little different.

Reasons to use the published SHA's, IMHO:
* **Users can more readily validate these against PyPi and check the source integrity**
* Maybe something to do with making it scraping-friendly (both for updating nixpkgs & any downstream).

File: pkgs/development/python-modules/werkzeug/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/werkzeug/default.nix:33]: I prefer to put ``pytestCheckHook`` first, but that's personal preference.

File: pkgs/development/python-modules/flask/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/flask/default.nix:45]: Missing maintainer :(

File: pkgs/development/python-modules/werkzeug/default.nix
Comment (drewrisinger) [inline pkgs/development/python-modules/werkzeug/default.nix:55]: Missing maintainer :(

**Comment by mweinelt** [comment]
Werkzeug optionally depends on watchdog, which requires https://developer.apple.com/documentation/coreservices/1455361-fseventstreameventflags/kfseventstreameventflagitemcloned, which requires macos-sdk 10.13+, we have 10.12 right now, so I made it conditional on `stdenv.isDarwin`.

@NixOS/darwin-maintainers

**Review by drewrisinger — CHANGES_REQUESTED** [review]
* [x] Diff: Maybe use PyPi hashes (see below comment)
* [x] Commits LGTM once the fixup is squashed
* [ ] Build in progress, will take a while... reviewing on x86_64-linux
* [x] Small build: built python3Packages.click & python3Packages.werkzeug cleanly.

Thanks for doing this, I was just looking at submitting a PR for click b/c I just saw it got an update.

File: pkgs/development/python-modules/werkzeug/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/werkzeug/default.nix:33]: Alphabetically sorted

File: pkgs/development/python-modules/click/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/click/default.nix:16]: Actually used `nix-update`, question is: Are you going to make the case for everyone touching these packages? Because there is no such policy in the python section. Maybe there should be, but then it should be discussed at a policy level.

**Comment by mweinelt** [comment]
Pushed the python2 package required to make nixops work again. Talked to @adisbladis who was confident that we can drop NixOps 1.7 before 21.11.

File: pkgs/development/python-modules/click/default.nix
Comment (drewrisinger) [reply pkgs/development/python-modules/click/default.nix:16]: I would probably make that case, but I don't feel strongly enough about it to hold up this particular PR.

Whether there's a policy or not, my de facto understanding was that ``fetchPypi`` sources use the PyPi Sha256, based on many auto-updates that it seems @FRidh does. I'm fine with it as-is for this PR

Comment (SuperSandro2000) [reply pkgs/development/python-modules/click/default.nix:16]: Those hashes are not base64 encoded and there is the possibility that nix calculates hashes different than pypi does. Doesn't matter in the end.

**Comment by SFrijters** [comment]
Note: the new 1.4.4 version of maestral (python3 only) now also requires click >= 8.

**Comment by mweinelt** [comment]
Merged into my local working tree for python-unstable.

**Comment by SFrijters** [comment]
@mweinelt What is the timeline for the Python batch upgrade? I might want to override `click` for the maestral packages if it will take a while.

**Comment by mweinelt** [comment]
@SFrijters Will start in an hour or so.

**Pull request closed without merging**
